parquet-variant/src/builder.rs

```diff
@@ -574,15 +575,16 @@ impl<'a> ListBuilder<'a> {
 /// A builder for creating [`Variant::Object`] values.
 ///
 /// See the examples on [`VariantBuilder`] for usage.
-pub struct ObjectBuilder<'a> {
+pub struct ObjectBuilder<'a, 'b> {
     parent_buffer: &'a mut ValueBuffer,
     metadata_builder: &'a mut MetadataBuilder,
     fields: BTreeMap<u32, usize>, // (field_id, offset)
     buffer: ValueBuffer,
-    pending: Option<(String, usize)>,
+    /// Is there a pending list or object that needs to be finalized?
+    pending: Option<(&'b str, usize)>,
```

Comment:
I started this PR to see if I could avoid this owned string, which it turns out you can